File: 2022/d1.py
sample = """1000
2000
3000

4000

5000
6000

7000
8000
9000

10000"""

data = [l.rstrip("\n") for l in open("d1.txt").readlines()]
snacks = []
curr = 0
for d in data:
    if d == "":
        snacks.append(curr)
        curr = 0
    else:
        curr += int(d)
snacks = list(sorted(snacks))
print(snacks[-1])
print(sum(snacks[-3:]))

File: 2022/d1.rb
all = []
curr = 0
File.open("d1.txt").readlines.each do |line|
	line = line.strip
	if line != ""
		curr += line.to_i
	else
		all << curr
		curr = 0
	end
end
all = all.sort
part1 = all[-1]
part2 = all[-3..-1].sum
puts part1
puts part2
